src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -118,6 +123,23 @@
     /** The angle at the start of the edge geometry. Internal representation like that of inAngle. */
     private byte outAngle;
 
+    /**
+     * The set of turn restrictions of this edge. Since most instances don't have any, we reuse
+     * a global instance in order to conserve memory.
+     *
+     * This field is optimized for low memory consumption and fast access, but modification is
+     * synchronized since it can happen concurrently.
+     *
+     * Why not use null to represent no turn restrictions?
+     * This would mean that the access would also need to be synchronized but since that is a very
+     * hot code path, it needs to be fast.
```

Comment:
I am not sure if this is true. A concurrent read is safe even if the variable is `null`, but we should avoid returning `null` for collections and instead always return an empty one. So, keep the non nullability of `turnRestrictions`, document with `@Nonnull`

Reply:
What I was trying to say is the following: If we keep null and don't synchronize the read, then another thread could set the variable to null before the getter returns.